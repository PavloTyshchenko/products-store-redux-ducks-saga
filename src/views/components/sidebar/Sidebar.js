import React from 'react';
import PropTypes from 'prop-types';

import { Button, ListGroup } from 'react-bootstrap';

import Search from './Search';
import Categories from './Categories';

export const Sidebar = ({ showClear, clearProducts }) => {

    return (
        <ListGroup data-test="sidebarComponent">
            <Search data-test="searchInput"/>
            <Categories data-test="categories"/>
            {
                showClear &&
                <Button data-test="clearButton"
                    variant="light"
                    className="mb-3"
                    onClick={clearProducts}>
                    Clear
                </Button>
            }
        </ListGroup>
    );
};

Sidebar.propTypes = {
    showClear: PropTypes.bool.isRequired,
    clearProducts: PropTypes.func.isRequired
};

Sidebar.defaultProps = {
    showClear: false,
    clearProducts: () => {}
};

export default Sidebar;
