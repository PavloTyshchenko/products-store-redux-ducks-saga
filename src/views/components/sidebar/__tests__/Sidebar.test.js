import React from 'react';
import { shallow } from 'enzyme';

import {Sidebar} from '../Sidebar';

const defaultProps = {
    showClear: false
};

const setup = (propsOption) => {
    const props = Object.assign(defaultProps, propsOption);
    const wrapper = shallow(<Sidebar {...props} />);
    return {
        wrapper,
        props
    };
};

describe('Sidebar Component', () => {

    it('Should render without errors', () => {
        const { wrapper } = setup();
        const component = wrapper.find(`[data-test='sidebarComponent']`);
        expect(component.length).toBe(1);
    });

    it('Should render a Search input', () => {
        const { wrapper } = setup();
        const search = wrapper.find(`[data-test='searchInput']`);
        expect(search.length).toBe(1);
    });

    it('Should render a Categories', () => {
        const { wrapper } = setup();
        const categories = wrapper.find(`[data-test='categories']`);
        expect(categories.length).toBe(1);
    });

    it('Should render a Clear Button with showClear is true', () => {
        const { wrapper } = setup({ showClear: true });
        const clearButton = wrapper.find(`[data-test='clearButton']`);
        expect(clearButton.length).toBe(1);
    });

    it('Should not render a Clear Button with showClear is false', () => {
        const { wrapper } = setup({ showClear: false });
        const clearButton = wrapper.find(`[data-test='clearButton']`);
        expect(clearButton.length).toBe(0);
    });

    test('Should match snapshot', () => {
        const component = shallow(<Sidebar />);
        expect(component.debug()).toMatchSnapshot();
    });
});