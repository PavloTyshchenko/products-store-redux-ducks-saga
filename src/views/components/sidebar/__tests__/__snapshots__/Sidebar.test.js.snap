// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component Should match snapshot 1`] = `
"<ForwardRef data-test=\\"sidebarComponent\\">
  <withRouter(Connect(Search)) data-test=\\"searchInput\\" />
  <withRouter(Connect(Categories)) data-test=\\"categories\\" />
</ForwardRef>"
`;
